optlang/gurobi_interface.py

```diff
@@ -371,22 +373,35 @@ def get_linear_coefficients(self, variables):
     def _get_expression(self):
         if self.problem is not None and self._expression_expired and len(self.problem._variables) > 0:
             grb_obj = self.problem.problem.getObjective()
-            terms = []
             variables = self.problem._variables
-            for i in range(grb_obj.size()):
-                terms.append(grb_obj.getCoeff(i) * variables[grb_obj.getVar(i).getAttr('VarName')])
-            expression = symbolics.add(terms)
-            # TODO implement quadratic objectives
-            self._expression = expression + getattr(self.problem, "_objective_offset", 0)
+            if self.problem.IsQP:
+                quadratic_expression = symbolics.add(
+                    [symbolics.Real(grb_obj.getCoeff(i)) *
+                     variables[grb_obj.getVar1(i).VarName] *
+                     variables[grb_obj.getVar2(i).VarName]
+                     for i in range(grb_obj.size())])
+                linear_objective = grb_obj.getLinExpr()
+            else:
+                quadratic_expression = symbolics.Real(0.0)
+                linear_objective = grb_obj
+            linear_expression = symbolics.add(
+                [symbolics.Real(linear_objective.getCoeff(i)) *
+                 variables[linear_objective.getVar(i).VarName]
+                 for i in range(linear_objective.size())]
+            )
+            self._expression = (linear_expression + quadratic_expression +
+                                getattr(self.problem, "_objective_offset", 0))
             self._expression_expired = False
         return self._expression
 
 
 @six.add_metaclass(inheritdocstring)
 class Configuration(interface.MathematicalProgrammingConfiguration):
-    def __init__(self, lp_method='primal', presolve=False, verbosity=0, timeout=None, *args, **kwargs):
+    def __init__(self, lp_method='primal', qp_method='barrier', presolve=False,
```

Comment:
If lp_method and qp_method overwrite each other I guess they should have the same default value.

Reply:
done